src/sound/instance.js

```diff
@@ -621,15 +677,18 @@ class SoundInstance extends EventHandler {
         this._manager.on('resume', this._onManagerResume, this);
         this._manager.on('destroy', this._onManagerDestroy, this);
 
-        // suspend immediately if manager is suspended
-        if (this._manager.suspended) {
-            this._onManagerSuspend();
-        }
-
-        if (!this._suspendInstanceEvents)
+        if (!this._suspendInstanceEvents) {
             this._onPlay();
+        }
 
-        return true;
+        if (this._pauseOnWaitingContextSuspension) {
```

Comment:
I am wondering about the mechanics of this bit - looks like this code could result in play & pause & stop being called on resume? Is it necessary to (possibly) call them all?

If you do need to call them all on resume, then presumably order matters? For example if the app attempted to play then pause then stop then play, presumably during resume you just want to play? (Where currently it'll play, then pause then stop the sound).

Seems like you may only need to track the current sound state and set that on resume? Sorry if I am misunderstanding everything!

Reply:
That makes total sense! Good callout, simplifies the code a lot!